MetaMask/metamask-extension
# avoid mutation of state in signTransaction action

## Explanation
Fixes #17745 

In #17508 i added store creation using reduxjs/toolkit instead of vanilla redux. One thing that comes with this is detection of mutation in state, which will throw an invariant error. This *should only be happening in dev mode* but its good to fix. This fixes the specific situation by cloning the transaction object before it used. 

## Screenshots/Screencaps



### Before



### After



## Manual Testing Steps
1. Create a transaction, but don't confirm it.
2. Edit transaction.
3. Change the recipient
4. On develop: Can't get to confirmation and invariant error in console.
5. On this branch: Confirmation.

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
